assembly/NFT/NFT.ts

```diff
@@ -0,0 +1,229 @@
+///////////////////////////////
+////////////////////////////
+////////////////////////
+// WIP allowance, and transferFrom Missing
+////////////////////////
+////////////////////////////
+///////////////////////////////
+
+import {Address, Storage, Context, generateEvent} from '../std';
+import {ByteArray} from '@massalabs/as/assembly';
+const ownerTokenKey: string = 'ownerOf_';
+const counterKey: string = 'Counter';
+const ownerKey: string = 'Owner';
+const baseURIKey: string = 'baseURI';
+const initCounter: u64 = 0;
+
+/**
+ * The NFT's main characteristics
+ */
+
+const _name: string = 'MASSA_NFT';
+const _symbol: string = 'NFT';
+const _maxSupply: string = '10000';
+const _baseURI: string = 'massa.net/nft/';
+
+/**
+ * Init the NFT with name, symbol, maxSupply and baseURI,
+ * init the counter to 0, the owner of the contract.
+ * @param {string} _ - unused see https://github.com/massalabs/massa-sc-std/issues/18
+ *
+ * @return {string}
+ */
+export function setNFT(_: string): string {
+  if (!Storage.has(counterKey)) {
+    Storage.set(baseURIKey, _baseURI);
```

Comment:
why you don't use setOf /getOf / hasOf ?

Reply:
why would you like store data in another address instead of the nft's ones ?